### PR TITLE
Add Effective_Pressure to NUCAPS reader

### PR DESCRIPTION
**Pull request by tommyjasmin**
- [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [ ] Fully documented 
 - [ ] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

**Comment by tommyjasmin** [comment]
@djhoese Dave - accidentally committed this to my master branch.  Maybe small enough change you can let it through, but talk to me if you prefer we undo and commit to a branch...

File: satpy/tests/reader_tests/test_nucaps.py [hunk 1]
```diff
@@ -208,6 +209,7 @@ def test_load_pressure_based(self):
         ])
         r.create_filehandlers(loadables)
         datasets = r.load(['Temperature',
+                           'Effective_Pressure',
```
Comment (djhoese) [inline satpy/tests/reader_tests/test_nucaps.py:212]: You'll need to update the total number of loaded datasets in the assertEqual below.

**Comment by tommyjasmin** [comment]
Yeah just noticed that (dataset is pressure-based), hopefully good now Dave.
TROPOMI work was on a separate branch and should stay so - easy to mentally conflate because we were working on an application using both NUCAPS and TROPOMI data.

File: satpy/tests/reader_tests/test_nucaps.py [hunk 2]
```diff
@@ -69,6 +69,7 @@ def get_test_content(self, filename, filename_info, filetype_info):
             ('Solar_Zenith', 'degrees', 'solar_zenith_angle'),
             ('Topography', 'meters', ''),
             ('Land_Fraction', '1', ''),
+            ('Effective_Pressure', 'mb', ''),
```
Comment (djhoese) [inline satpy/tests/reader_tests/test_nucaps.py:72]: This should be moved to the `for` loop below so the test data is pressure based.

**Comment by djhoese** [comment]
Looks good. Thanks @tommyjasmin!

**Pull request merged**